src/analysis/Environments/DeleteButton.ts

```diff
@@ -1,36 +1,46 @@
 import _ from 'lodash/fp';
 import { ReactNode } from 'react';
 import { h } from 'react-hyperscript-helpers';
-import { isResourceDeletable } from 'src/analysis/utils/resource-utils';
 import { getDisplayRuntimeStatus } from 'src/analysis/utils/runtime-utils';
 import { Link } from 'src/components/common';
 import { makeMenuIcon } from 'src/components/PopupTrigger';
 import { isApp } from 'src/libs/ajax/leonardo/models/app-models';
 import { LeoRuntimeStatus } from 'src/libs/ajax/leonardo/models/runtime-models';
+import * as Utils from 'src/libs/utils';
 
-import { DecoratedComputeResource } from './Environments.models';
+import { DecoratedComputeResource, PermissionsAndStateProvider } from './Environments.models';
+
+type DeletePermissionsProvider = Pick<PermissionsAndStateProvider, 'canDeleteApp' | 'canDeleteResource'>;
 
 export interface DeleteButtonProps {
   resource: DecoratedComputeResource;
+  permissions: DeletePermissionsProvider;
   onClick: (resource: DecoratedComputeResource) => void;
 }
 
 export const DeleteButton = (props: DeleteButtonProps): ReactNode => {
-  const { resource, onClick } = props;
+  const { resource, permissions, onClick } = props;
   const resourceType = isApp(resource) ? 'app' : 'runtime';
-  const isDeletable = isResourceDeletable(resourceType, resource);
+  const isDeletable =
+    resourceType === 'app' ? permissions.canDeleteApp(resource) : permissions.canDeleteResource(resourceType, resource);
 
   return h(
     Link,
     {
       disabled: !isDeletable,
-      tooltip: isDeletable
-        ? 'Delete cloud environment'
-        : `Cannot delete a cloud environment while in status ${_.upperCase(
-            getDisplayRuntimeStatus(resource.status as LeoRuntimeStatus)
-          )}.`,
+      tooltip: Utils.cond(
+        [isDeletable, () => 'Delete cloud environment'],
+        [!permissions.canDeleteApp(resource), () => 'Deleting not yet supported'],
+        [
+          Utils.DEFAULT,
+          () =>
+            `Cannot delete a cloud environment while in status ${_.upperCase(
+              getDisplayRuntimeStatus(resource.status as LeoRuntimeStatus)
+            )}.`,
+        ]
+      ),
       onClick: () => onClick(resource),
     },
-    [makeMenuIcon('trash'), 'Delete']
```

Comment:
What does this `undefined` do?

Reply:
At the moment, nothing. TS was unhappy here because it expected a `props` arg, but there wasn't one. I added the `undefined` to make this function happy.